cartridge/roles/metrics.lua

```diff
@@ -138,6 +143,10 @@ local function apply_config(conf)
         end
         metrics_config_present = false
     end
+    metrics.clear()
```

Comment:
I can already see a tons of people (including myself) that got their custom metrics broken because of code design (like storing collector in a variable because searching it in O(N)-registry each time is painful). Moreover, what's about custom global labels and callbacks? It's not obvious how to write `clear()`-proof custom metrics code. I'm not sure that this is even possible. At least we never insisted on it and didn't have any examples and guides on how to do it. Casually adding this line of code will bring a great disaster

Reply:
Removed `metrics.clear` call